src/main.rs

```diff
@@ -32,46 +34,47 @@ use window_manager::{Window, WindowManager};
 #[clap(version, about)]
 struct Args;
 
-fn pretty_window(
-    window: &Window,
-    icon_mappings: &[(String, String)],
-    fallback_icon: &str,
-) -> String {
-    for (name, icon) in icon_mappings {
+fn pretty_window(config: &Config, window: &Window) -> String {
+    for (name, icon) in &config.mappings {
         if window.matches(name) {
             return icon.clone();
         }
     }
-    error!("Couldn't identify window: {window:?}");
+    warn!("Couldn't identify window: {window:?}");
     info!("Make sure to add an icon for this file in your config file!");
-    fallback_icon.to_string()
+    config.fallback_icon().into()
 }
 
-fn pretty_windows(
-    windows: &[Window],
-    icon_mappings: &[(String, String)],
-    fallback_icon: &str,
-) -> String {
+fn pretty_windows(config: &Config, windows: &[Window]) -> String {
     let mut s = String::new();
-    for window in windows {
-        s.push_str(&pretty_window(window, icon_mappings, fallback_icon));
-        s.push(' ');
+    if config.other.merge {
+        let mut set = BTreeSet::new();
+        for window in windows {
+            set.insert(pretty_window(config, window));
```

Comment:
I have a slight problem with this implementation from a UX perspective:
since we're using a BTreeSet, the windows will end up sorted alphabetically regardless of their position on the string.
In the non-merged version, the order of the windows matches their position on the string, which I think is nice.
Now in the merged version, the order will appear arbitrary in the simple case.
For instance, for a workspace with one firefox and one alacritty window with the default config, the name will be: """, even if the browser is on the left half of the screen.

I think a nicer way to handle this would be to use a `Vec` like before, but maybe collect the already seen names in a `HashSet` (or `BTreeSet`. It doesn't matter. We don't need ordering) as we iterate through them and only append unseen ones. That way, we will list them by first order of appearance which I feel is nicer from a user's point of view.